qcodes/instrument/base.py

```diff
@@ -502,12 +502,9 @@ def __repr__(self) -> str:
     def __del__(self) -> None:
         """Close the instrument and remove its instance record."""
         try:
-            wr = weakref.ref(self)
-            if wr in getattr(self, '_instances', []):
-                self._instances.remove(wr)
             self.close()
-        except:
-            pass
+        except BaseException:
+            log.exception("Failed to close instrument in destructor")
```

Comment:
I am not sure if logging from a ``_del__`` method is safe? Perhaps we should just ommit this

Reply:
agree to omit. close already has logging